### PR TITLE
Task/datetime block template

### PR DESCRIPTION
**Pull request by paulmskim**
**Ticket:** [TEC-3349]

Do not merge until QA passes.

This is based off of https://github.com/moderntribe/the-events-calendar/pull/3129 currently because it depends on https://github.com/moderntribe/the-events-calendar/pull/3129. This PR will be rebased to the bucket branch once https://github.com/moderntribe/the-events-calendar/pull/3129 is merged. The diff will be largely the same.

This PR works on the block editor data management for TEC ([TEC-3021]). The specific block affected include the datetime block.

[TEC-3349]: https://moderntribe.atlassian.net/browse/TEC-3349
[TEC-3021]: https://moderntribe.atlassian.net/browse/TEC-3021

### CODE REVIEW

File: src/Tribe/Editor/Blocks/Event_Datetime.php [hunk 1]
```diff
@@ -50,4 +50,81 @@ public function assets() {
...
+				],
+				'showTimeZone'  => [
+					'type'    => 'boolean',
+					'default' => tribe_get_option( 'tribe_events_timezones_show_zone', false ),
+				],
+				'timeZoneLabel' => [
+					'type'    => 'string',
+					'default' => class_exists( 'Tribe__Timezones' ) ? Tribe__Timezones::wp_timezone_string() : get_option( 'timezone_string', 'UTC' ),
+				],
+				// Only available for classic users
```
Comment (tr1b0t) [inline src/Tribe/Editor/Blocks/Event_Datetime.php:103]: :no_entry_sign: Inline comments must end in full-stops, exclamation marks, or question marks

<sub>hash: 5b6e375ed3c0cc105a85a7bf749cd9a2</sub>

File: src/Tribe/Editor/Configuration.php [hunk 2]
```diff
@@ -39,6 +40,14 @@ public function editor_config( $editor_config ) {
 			]
 		);
 
+		$blocks = empty( $editor_config['blocks'] ) ? [] : $editor_config['blocks'];
```
Comment (tr1b0t) [inline src/Tribe/Editor/Configuration.php:43]: :warning: Equals sign not aligned with surrounding assignments; expected 18 spaces but found 1 space

<sub>hash: 68d393df09fb6ad436577ae4f1ea129a</sub>

**Comment by paulmskim** [comment]
@lucatume running into a test issue here, I updated wp-browser and the codeception modules in composer.json but it doesn't seem to be the issue.

File: src/modules/blocks/event-datetime/content/template.js [hunk 3]
```diff
@@ -176,68 +176,55 @@ const renderTimezone = () => {
 const renderExtras = ( props ) => {
```
Comment (alliuca) [inline src/modules/blocks/event-datetime/content/template.js:176]:
```suggestion
const renderExtras = ( props ) => (
	<Fragment>
		{ renderTimezone( props ) }
		{ renderPrice( props ) }
	</Fragment>
);
```

File: src/modules/blocks/event-datetime/dashboard/template.js [hunk 4]
```diff
@@ -43,13 +40,14 @@ const {
...
 } = momentUtil;
+const { editorConstants, settings, wpHooks } = globals;
 
-FORMATS.date = globals.settings() && globals.settings().dateWithYearFormat
-	? globals.settings().dateWithYearFormat
+FORMATS.date = settings() && settings().dateWithYearFormat
+	? settings().dateWithYearFormat
 	: __( 'F j', 'the-events-calendar' );
 
 const shouldHideUpsell = () => {
```
Comment (alliuca) [inline src/modules/blocks/event-datetime/dashboard/template.js:49]:
```suggestion
const shouldHideUpsell = () => editorConstants().hideUpsell;
```

File: src/modules/blocks/event-datetime/dashboard/template.js [hunk 5]
```diff
@@ -171,53 +169,41 @@ class Calendars extends PureComponent {
...
+					<Calendars { ...props } />
+				</section>
+				<footer className="tribe-editor__subtitle__footer">
+					<div className="tribe-editor__subtitle__footer-date">
+						<div className="tribe-editor__subtitle__time-pickers">
+							{ renderStartTimePicker( props ) }
+							{
+								( multiDay || ! allDay ) && (
+									<span className={ classNames( 'tribe-editor__separator', 'tribe-editor__time-picker__separator' ) }>
+										{ ' '.concat( separatorTime, ' ' ) }
```
Comment (alliuca) [inline src/modules/blocks/event-datetime/dashboard/template.js:192]:
```suggestion
										{ ` ${separatorTime} `}
```

**Pull request merged**